<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"zhengxingsong.github.io","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":true,"show_result":true,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":true,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="最新版发布页面：https:&#x2F;&#x2F;github.com&#x2F;shenwei356&#x2F;seqkit&#x2F;releases 安装目前可以安装比较最新的版本Version: 至少 &gt;&#x3D; 0.15.0。seqkit在github上的维护力度也比较大，功能比较完善，放心使用。">
<meta property="og:type" content="article">
<meta property="og:title" content="seqkit">
<meta property="og:url" content="http://zhengxingsong.github.io/%E7%94%9F%E4%BF%A1/seqkit/index.html">
<meta property="og:site_name" content="阿狒的博客">
<meta property="og:description" content="最新版发布页面：https:&#x2F;&#x2F;github.com&#x2F;shenwei356&#x2F;seqkit&#x2F;releases 安装目前可以安装比较最新的版本Version: 至少 &gt;&#x3D; 0.15.0。seqkit在github上的维护力度也比较大，功能比较完善，放心使用。">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2021-11-10T15:54:30.000Z">
<meta property="article:modified_time" content="2021-11-10T15:55:48.056Z">
<meta property="article:author" content="阿狒">
<meta property="article:tag" content="生信">
<meta property="article:tag" content="biosoft">
<meta property="article:tag" content="CommonSoft">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="http://zhengxingsong.github.io/%E7%94%9F%E4%BF%A1/seqkit/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>

  <title>seqkit | 阿狒的博客</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">阿狒的博客</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">愿热爱生活的人可以不被辜负</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>标签<span class="badge">9</span></a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>分类<span class="badge">4</span></a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档<span class="badge">7</span></a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>
  <div class="reading-progress-bar"></div>
  <a role="button" class="book-mark-link book-mark-link-fixed"></a>

  <a href="https://github.com/zhengxingSong" class="github-corner" title="Follow me on GitHub" aria-label="Follow me on GitHub" rel="noopener" target="_blank"><svg width="80" height="80" viewBox="0 0 250 250" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="http://zhengxingsong.github.io/%E7%94%9F%E4%BF%A1/seqkit/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/header.jpg">
      <meta itemprop="name" content="阿狒">
      <meta itemprop="description" content="人这一生总会死去，你不妨大胆一些，爱一个人，攀一座山，追一个梦，是的，你不妨大胆一些。">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="阿狒的博客">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          seqkit
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">发表于</span>
              

              <time title="创建时间：2021-11-10 23:54:30 / 修改时间：23:55:48" itemprop="dateCreated datePublished" datetime="2021-11-10T23:54:30+08:00">2021-11-10</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/%E7%94%9F%E4%BF%A1/" itemprop="url" rel="index"><span itemprop="name">生信</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="阅读次数" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">阅读次数：</span>
              <span id="busuanzi_value_page_pv"></span>
            </span><br>
            <span class="post-meta-item" title="本文字数">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
                <span class="post-meta-item-text">本文字数：</span>
              <span>17k</span>
            </span>
            <span class="post-meta-item" title="阅读时长">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
                <span class="post-meta-item-text">阅读时长 &asymp;</span>
              <span>16 分钟</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>最新版发布页面：<a target="_blank" rel="noopener" href="https://github.com/shenwei356/seqkit/releases">https://github.com/shenwei356/seqkit/releases</a></p>
<h2 id="安装"><a href="#安装" class="headerlink" title="安装"></a>安装</h2><p>目前可以安装比较最新的版本Version: 至少 &gt;= 0.15.0。seqkit在github上的维护力度也比较大，功能比较完善，放心使用。</p>
<span id="more"></span>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda seqkit</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 补充bedops(gtf2bed)和csvtk工具(可选)</span></span><br><span class="line">conda install -c bioconda bedops -y</span><br><span class="line">conda install -c bioconda csvtk -y</span><br></pre></td></tr></table></figure>

<p>可选在[发布页][ <a target="_blank" rel="noopener" href="https://github.com/shenwei356/seqkit/releases">https://github.com/shenwei356/seqkit/releases</a> ]直接下载适合的版本。</p>
<p>seqkit 一共有37个可用的命令，详细内容如下：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><span class="line">amplicon        通过引物检索扩增子(或其周围的特定区域)</span><br><span class="line">bam             检查和在线绘制BAM记录文件的直方图</span><br><span class="line">common          通过id/名称/序列查找多个文件的公共序列</span><br><span class="line">concat          连接多个文件中具有相同ID的序列</span><br><span class="line">convert         转换FASTQ质量编码格式：支持格式包括：桑格，Solexa和Illumina</span><br><span class="line">duplicate       重复序列N次</span><br><span class="line">faidx           创建FASTA索引文件并提取子序列</span><br><span class="line">fish            使用局部比对在较大的序列中寻找短序列</span><br><span class="line">fq2fa           转换FASTQ到FASTA</span><br><span class="line">fx2tab          将FASTA/Q转换为表格格式(包含长度/GC含量/GC偏好)</span><br><span class="line">genautocomplete 生成shell自动完成脚本</span><br><span class="line">grep            通过ID/name/sequence/sequence motif搜索序列，允许错配</span><br><span class="line">head            打印第一条序列</span><br><span class="line">help            打印帮助信息</span><br><span class="line">locate          定位序列，或者motifs，允许错配</span><br><span class="line">mutate          编辑序列(点突变、插入、删除)</span><br><span class="line">pair            匹配双端序列文件</span><br><span class="line">range           打印一个范围内的序列</span><br><span class="line">rename          重命名重复序列ID</span><br><span class="line">replace         使用正则表达式修改名称或者序列</span><br><span class="line">restart         重置环状基因组的起始位置</span><br><span class="line">rmdup           通过id/名称/序列删除重复的序列</span><br><span class="line">sample          按数量或比例对序列进行抽样</span><br><span class="line">sana            清理损坏的单行fastq文件</span><br><span class="line">scat            real time recursive concatenation and streaming of fastx files</span><br><span class="line">seq             转换序列(反向，补充，提取ID…)</span><br><span class="line">shuffle         随机序列</span><br><span class="line">sliding         序列滑窗提取，支持环形基因组</span><br><span class="line">sort            按id/名称/序列/长度排序序列</span><br><span class="line">split           按id/seq区域/大小/部件将序列拆分为文件(主要用于FASTA)</span><br><span class="line">split2          按序列数量/文件数将序列拆分为多个文件(FASTA, PE/SE FASTQ)</span><br><span class="line">stats           FASTA/Q文件的简单统计</span><br><span class="line">subseq          通过region/gtf/bed得到子序列，包括侧翼序列</span><br><span class="line">tab2fx          转换表格格式为FASTA/Q格式</span><br><span class="line">translate       翻译DNA/RNA到蛋白质序列(支持歧义碱基)</span><br><span class="line">version         打印版本信息并检查是否更新</span><br><span class="line">watch           序列特征的监测和在线直方图</span><br></pre></td></tr></table></figure>

<p>参数</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">Flags:</span><br><span class="line">      --alphabet-guess-seq-length int   seqkit根据第一个FASTA记录猜测序列类型的序列前缀的长度(0表示整个序列)(默认10000)</span><br><span class="line">  -h, --help                            显示帮助</span><br><span class="line">      --id-ncbi                         FASTA头是ncbi风格的，例如&gt;gi|110645304|ref|NC_002516.2 </span><br><span class="line">      --id-regexp string                用于解析ID的正则表达式(default &quot;^(\\S+)\\s?&quot;)，匹配空格前的部分为序列名</span><br><span class="line">      --infile-list string              输入文件列表中的文件 (one file per line), if given, they are appended to files from cli arguments</span><br><span class="line">  -w, --line-width int                  输出FASTA格式时的行宽 (0 for no wrap) (default 60)</span><br><span class="line">  -o, --out-file string                 输出 (&quot;-&quot; for stdout, suffix .gz for gzipped out) (default &quot;-&quot;) -代表标准输出，加.gz可输出压缩文件</span><br><span class="line">      --quiet                           保持安静，不要显示额外的信息</span><br><span class="line">  -t, --seq-type string                 序列类型 (dna|rna|protein|unlimit|auto) (auto, 按第一个序列自动检测) (default &quot;auto&quot;)</span><br><span class="line">  -j, --threads int                     CPU数量 (默认单核为1，多核为2) (default 2)</span><br></pre></td></tr></table></figure>

<h2 id="常用功能"><a href="#常用功能" class="headerlink" title="常用功能"></a>常用功能</h2><h3 id="1-根据gtf构建bed文件"><a href="#1-根据gtf构建bed文件" class="headerlink" title="1.根据gtf构建bed文件"></a>1.根据gtf构建bed文件</h3><p>提取1号染色体序列及注释作为示例</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">seqkit grep -p 1 Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz -o chr1.fa.gz</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 注释信息按照染色体取子集：提取第一条染色体的基因组注释信息：<span class="string">&#x27;^1&#x27;</span></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用gtf文件提取</span></span><br><span class="line">zcat Arabidopsis_thaliana.TAIR10.49.gtf.gz | grep -w &#x27;^1&#x27; | gzip -c &gt; chr1.gtf.gz</span><br><span class="line"><span class="meta">#</span><span class="bash"> 提取第一条染色体的bed文件，用法相同</span></span><br><span class="line">zcat Arabidopsis_thaliana.TAIR10.49.bed.gz | grep -w &#x27;^1&#x27; | gzip -c &gt; chr1.bed.gz</span><br></pre></td></tr></table></figure>

<h3 id="2-stats-FASTA-Q文件的简单统计"><a href="#2-stats-FASTA-Q文件的简单统计" class="headerlink" title="2. stats FASTA/Q文件的简单统计"></a>2. stats FASTA/Q文件的简单统计</h3><p>统计序列格式fasta(fa)/fastq(fq)、内容类型DNA/RNA/Protein，序列数量、总长度，最小、平均和最大长度</p>
<p>下面就四种格式序列构建和简单统计。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> FASTA DNA</span></span><br><span class="line">echo -e &quot;&gt;seq\nacgtryswkmbdhvACGTRYSWKMBDHV&quot; | seqkit stats</span><br><span class="line"><span class="meta">#</span><span class="bash"> RNA</span> </span><br><span class="line">echo -e &quot;&gt;seq\nACGUN\nACGUN&quot; | seqkit stats</span><br><span class="line"><span class="meta">#</span><span class="bash"> Protein</span></span><br><span class="line">echo -e &quot;&gt;seq\nabcdefghijklmnpqrstvwyz&quot; | seqkit stats</span><br><span class="line"><span class="meta">#</span><span class="bash"> FASTQ DNA</span></span><br><span class="line">echo -e &quot;@read\nACTGCN\n+\n@IICCG&quot; | seqkit stats</span><br></pre></td></tr></table></figure>

<p>使用fq或者fa文件进行演示</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"># 一般模式</span><br><span class="line">seqkit stats C1_1.fq.gz</span><br><span class="line">#--输出结果tab分隔</span><br><span class="line">seqkit stats C1_1.fq.gz -T</span><br><span class="line">#--输出文件转化其他格式</span><br><span class="line">seqkit stats C1_1.fq.gz -T| csvtk pretty -t</span><br><span class="line">seqkit stats C1_1.fq.gz -T| csvtk csv2md -t</span><br><span class="line"># 统计更多信息 -a</span><br><span class="line">seqkit stats C1_1.fq.gz -a</span><br><span class="line"># j多线程加速，尤其是对于具有多个序列文件会加速</span><br><span class="line"># seqkit stats -j 2 *.fq.gz</span><br></pre></td></tr></table></figure>

<h3 id="3-seq-转换序列-（反向、互补-提取ID）"><a href="#3-seq-转换序列-（反向、互补-提取ID）" class="headerlink" title="3. seq 转换序列 （反向、互补/提取ID）"></a>3. seq 转换序列 （反向、互补/提取ID）</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit seq [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">-p, --complement # 取互补序列</span><br><span class="line">--dna2rna # DNA到RNA--rna2dna RNA到DNA</span><br><span class="line">-G, --gap-letters # string  gap letters (default &quot;- \t.&quot;),字符间隔符</span><br><span class="line">-l, --lower-case # 用小写字母打印序列</span><br><span class="line">-M, --max-len int # 只打印短于最大长度的的序列</span><br><span class="line">-n, --name # 只打印name</span><br><span class="line">-g, --remove-gaps # 移去组装序列中的gap</span><br><span class="line">-r, --reverse # 取反向序列</span><br><span class="line">-i, --only-id # 只打印ID而不是全名</span><br><span class="line">-q, --qual # 只序列质量</span><br><span class="line">-s, --seq # 只打印序列</span><br><span class="line">--id-regexp # string 对于解析ID的正则表达式，(default &quot;^(\\S+)\\s?&quot;)</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">“-n”: 提取序列ID，包括“&gt;”后面的全部内容</span><br><span class="line">“-n -i”: 仅提取第一个空格前的ID</span><br></pre></td></tr></table></figure>

<h4 id="3-1-按长度过滤-常用"><a href="#3-1-按长度过滤-常用" class="headerlink" title="3.1 按长度过滤(常用)"></a>3.1 按长度过滤(常用)</h4><p>扩增子分析时要筛选扩增长度相近的片段，过长或过短一般都要删除。宏基因组中比如组装的结果，经常要过滤&lt;200/300bp的短片段，分箱时要筛选&gt;1000/2000的长片段使用。本条命令非常多的应用场景。筛选后结果可用 &gt; 写入文件</p>
<ul>
<li>-m 按照序列长度过滤，表示保留的最小长度，-M 此为保留的最大长度</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">--提取序列长度大于60的并统计长度信息</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit seq -m 60 | seqkit stats</span><br><span class="line"><span class="meta">#</span><span class="bash"> 设置最小序列长度和最大序列长度，用于过滤序列，并统计</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit seq -m 100 -M 1000 | seqkit stats</span><br><span class="line"><span class="meta">#</span><span class="bash"> 保存&gt;100且&lt;1000长度的序列</span></span><br><span class="line">seqkit seq -m 100 -M 1000 hairpin.fa.gz &gt; hairpin100-1000.fa</span><br><span class="line">seqkit stat hairpin100-1000.fa</span><br></pre></td></tr></table></figure>

<h4 id="3-2-提取ID"><a href="#3-2-提取ID" class="headerlink" title="3.2 提取ID"></a>3.2 提取ID</h4><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">head gene.fa</span><br><span class="line"><span class="meta">#</span><span class="bash"><span class="comment"># 名称全行</span></span></span><br><span class="line">seqkit seq gene.fa -n | head</span><br><span class="line"><span class="meta">#</span><span class="bash"> 仅仅打印ID</span></span><br><span class="line">seqkit seq gene.fa -n -i | head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用正则表达式提取名字中的信息</span></span><br><span class="line">zcat hairpin.fa.gz | head</span><br><span class="line"><span class="meta">#</span><span class="bash"> 提取ID中第二个字段作为ID</span></span><br><span class="line">seqkit seq hairpin.fa.gz -n -i --id-regexp &quot;^[^\s]+\s([^\s]+)\s&quot; | head</span><br></pre></td></tr></table></figure>

<h4 id="3-3-单行-多行转换"><a href="#3-3-单行-多行转换" class="headerlink" title="3.3 单行/多行转换"></a>3.3 单行/多行转换</h4><ul>
<li>-s提取并展示序列</li>
<li>-w 代表每行的碱基数量，0代表不换行</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">仅仅提取序列 -s</span></span><br><span class="line">seqkit seq gene.fa -s -w 0|head</span><br><span class="line"><span class="meta">#</span><span class="bash">--将多行序列转化为标准4行FASTQ</span></span><br><span class="line">seqkit seq C1_1.fq.gz -w 0|head</span><br></pre></td></tr></table></figure>

<h4 id="3-4-反向-互补"><a href="#3-4-反向-互补" class="headerlink" title="3.4 反向/互补"></a>3.4 反向/互补</h4><ul>
<li>-r 序列反向</li>
<li>-p序列互补</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 序列反向互补,-r反向，-p互补</span></span><br><span class="line">seqkit seq hairpin.fa.gz -r -p|head</span><br></pre></td></tr></table></figure>

<h4 id="3-5-删除gap-大小写转换"><a href="#3-5-删除gap-大小写转换" class="headerlink" title="3.5 删除gap/大小写转换"></a>3.5 删除gap/大小写转换</h4><ul>
<li>-g 去除序列中的间隔,将中间的横杠去掉</li>
<li>-u转化序列为大写字母展示</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">echo -e &quot;&gt;seq\nACGT-ACTGC-acc&quot; | seqkit seq -g -u</span><br></pre></td></tr></table></figure>

<h4 id="3-6-RNA转为DNA"><a href="#3-6-RNA转为DNA" class="headerlink" title="3.6 RNA转为DNA"></a>3.6 RNA转为DNA</h4><ul>
<li>—rna2dna 将RNA序列转化为DNA序列</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">echo -e &quot;&gt;seq\nUCAUAUGCUUGUCUCAAAGAUUA&quot; | seqkit seq --rna2dna</span><br></pre></td></tr></table></figure>

<h3 id="4-subseq通过指定区域"><a href="#4-subseq通过指定区域" class="headerlink" title="4. subseq通过指定区域"></a>4. subseq通过指定区域</h3><ul>
<li>-r 通过区域来截取序列</li>
</ul>
<p>如1:12提取前12个碱基，-12:-1提取序列结尾12个碱基；<br>for last 12 bases, 13:-1 for cutting first 12 bases. type “seqkit subseq -h” for more examples</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">-提取序列前1：12个碱基</span></span><br><span class="line">zcat C1_1.fq.gz | seqkit subseq -r 1:12 |head</span><br><span class="line"><span class="meta">#</span><span class="bash">-提取序列最后1：12个碱基</span></span><br><span class="line">zcat C1_1.fq.gz | seqkit subseq -r -12:-1 |head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash">取第12至倒数第12个碱基，即前11和后11个碱基去掉</span></span><br><span class="line">zcat C1_1.fq.gz | seqkit subseq -r 12:-12| head</span><br></pre></td></tr></table></figure>

<p>基于gtf/bed信息挑选子序列。</p>
<ul>
<li>—gtf  根据gtf文件挑选基因，这部分功能用于根据基因注释快速提取基因序列，在宏基因组、转录组、重测序中常用。—chr 选择染色体，—feature cds选择序列类型</li>
</ul>
<p>以拟南芥基因组的序列和注释数据演示：提取第一条染色体上的CDS基因信息，并统计基本信息</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">seqkit subseq --gtf Arabidopsis_thaliana.TAIR10.49.gtf.gz --chr 1 --feature cds  Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz  &gt; chr1.gtf.cds.fa</span><br><span class="line">seqkit stats chr1.gtf.cds.fa</span><br></pre></td></tr></table></figure>

<ul>
<li>-u 可以提取目标基因上游的序列</li>
<li>-f 目标区域不展示</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">--挑选序列并多加上上游的3个碱基</span></span><br><span class="line">seqkit subseq --gtf Arabidopsis_thaliana.TAIR10.49.gtf.gz Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz -u 3 |head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 仅提取上游序列，如提取启动子区2k：-f仅定位不输出位置序列，-u输出上游序列，此处示例3bp</span></span><br><span class="line">seqkit subseq --gtf Arabidopsis_thaliana.TAIR10.49.gtf.gz Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz -u 3 -f |head</span><br></pre></td></tr></table></figure>

<h3 id="5-sliding-滑窗提取序列，支持环状基因组"><a href="#5-sliding-滑窗提取序列，支持环状基因组" class="headerlink" title="5. sliding 滑窗提取序列，支持环状基因组"></a>5. sliding 滑窗提取序列，支持环状基因组</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">-s 步长为3，-W 序列长度为6个碱基</span></span><br><span class="line">echo -e &quot;&gt;seq\nACGTacgtNN&quot; | seqkit sliding -s 3 -W 6</span><br><span class="line"><span class="meta">#</span><span class="bash"> -g 贪婪模式，后面不足6个那也取</span></span><br><span class="line">echo -e &quot;&gt;seq\nACGTacgtNN&quot; | seqkit sliding -s 3 -W 6 -g</span><br><span class="line"><span class="meta">#</span><span class="bash"> 环状DNA模式-C，首尾不算中断，环状</span></span><br><span class="line">echo -e &quot;&gt;seq\nACGTacgtNN&quot; | seqkit sliding -s 3 -W 6 -C</span><br></pre></td></tr></table></figure>

<p>步长为5，取30个碱基序列，然后统计GC含量</p>
<ul>
<li>fx2tab：统计fasta/fastq序列的信息为表格</li>
<li>-n仅输出ID，不输出序列</li>
<li>-g为GC含量</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">zcat hairpin.fa.gz | seqkit sliding -s 5 -W 30 | seqkit fx2tab -n -g |head</span><br></pre></td></tr></table></figure>

<h3 id="6-faidx-创建FASTA索引文件并提取子序列"><a href="#6-faidx-创建FASTA索引文件并提取子序列" class="headerlink" title="6. faidx 创建FASTA索引文件并提取子序列"></a>6. faidx 创建FASTA索引文件并提取子序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 解压</span></span><br><span class="line">zcat hairpin.fa.gz &gt; hairpin.fa</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 建索引*.fai文件</span></span><br><span class="line">seqkit faidx hairpin.fa</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> ID信息：hsa-let-7a-1 hsa-let-7a-2</span></span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1 hsa-let-7a-2</span><br><span class="line"><span class="meta">#</span><span class="bash"> -f 标题行全部显示</span></span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1 hsa-let-7a-2 -f</span><br><span class="line"><span class="meta">#</span><span class="bash"> 提取序列并选择区域显示</span></span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1:1-10</span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1:-10--1</span><br><span class="line">seqkit faidx hairpin.fa hsa-let-7a-1:1</span><br><span class="line"><span class="meta">#</span><span class="bash"> 检查hsa开头的序列并统计</span></span><br><span class="line">seqkit faidx hairpin.fa hsa -r | seqkit stats</span><br></pre></td></tr></table></figure>

<h3 id="7-watch-序列质量的监测和在线直方图"><a href="#7-watch-序列质量的监测和在线直方图" class="headerlink" title="7. watch 序列质量的监测和在线直方图"></a>7. watch 序列质量的监测和在线直方图</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">-取对数展示直方图</span></span><br><span class="line">seqkit watch -L -f ReadLen hairpin.fa</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 每五千个做一个图保存在pdf文件中</span></span><br><span class="line">seqkit watch -p 500000 -O qhist.pdf -f MeanQual C1_1.fq.gz</span><br></pre></td></tr></table></figure>

<ul>
<li>从有错误记录的fastq文件中挽救可用的读取</li>
</ul>
<h3 id="8-sana：清理损坏fastq文件"><a href="#8-sana：清理损坏fastq文件" class="headerlink" title="8. sana：清理损坏fastq文件"></a>8. sana：清理损坏fastq文件</h3><p>这里我专门将C1_1.fq的第一个序列进行了错位，进行测试。这个操作往往在进行数据整合的时候可以有很大作用。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">zcat C1_1.fq.gz|sed &#x27;2 s/^/A/&#x27; &gt; C1_1_bad.fq</span><br><span class="line">seqkit sana C1_1_bad.fq -o rescued.fq.gz</span><br></pre></td></tr></table></figure>

<h3 id="9-fq2fa-将fq转为fa格式"><a href="#9-fq2fa-将fq转为fa格式" class="headerlink" title="9. fq2fa 将fq转为fa格式"></a>9. fq2fa 将fq转为fa格式</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit fq2fa C1_1.fq.gz -o C1_1.fa</span><br></pre></td></tr></table></figure>

<h3 id="10-fx2tab-amp-tab2fx-序列转化表格格式"><a href="#10-fx2tab-amp-tab2fx-序列转化表格格式" class="headerlink" title="10. fx2tab &amp; tab2fx 序列转化表格格式"></a>10. fx2tab &amp; tab2fx 序列转化表格格式</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit fx2tab [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">-a, --alphabet 打印字母表字母</span><br><span class="line">-q, --avg-qual 打印read的平均质量</span><br><span class="line">-B, --base-content strings 输出指定的碱基含量</span><br><span class="line">-g, --gc 输出GC含量</span><br><span class="line">-H, --header-line 打印标题列</span><br><span class="line">-l, --length 打印序列的长度</span><br><span class="line">-n, --name 只打印名字，而没有序列或者质量</span><br><span class="line">-i, --only-id 只打印基因的ID</span><br></pre></td></tr></table></figure>

<p>这一转化很有用，往往用于表格/矩阵处理的时候。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit fx2tab hairpin.fa.gz | head -n 2</span><br></pre></td></tr></table></figure>

<p>通过矩阵格式的序列文件统计序列长度和质量值</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 打印序列长度、GC含量</span></span><br><span class="line">seqkit fx2tab hairpin.fa.gz -l -g -n -i -H | head</span><br></pre></td></tr></table></figure>

<ul>
<li>tab2fx 和表格格式转化为序列格式</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> seqkit tab2fx 表格形式转化为序列形式</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit fx2tab | seqkit tab2fx | head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 转化为表格，然后排序，然后转化回去</span></span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit fx2tab -l \</span><br><span class="line">    | sort -t &quot;`echo -e &#x27;\t&#x27;`&quot; -n -k4,4 \</span><br><span class="line">    | seqkit tab2fx | head</span><br><span class="line"><span class="meta">#</span><span class="bash"> 等同于下面的命令</span></span><br><span class="line">seqkit sort -l hairpin.fa.gz | head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 通过这个转化可以将很多在表格中实现的数据处理方法用于序列</span></span><br><span class="line">例如下面的提取1000个序列：seqkit fx2tab hairpin.fa.gz | head -n 1000 | seqkit tab2fx | head</span><br></pre></td></tr></table></figure>

<h3 id="11-translate-翻译DNA-RNA为蛋白质序列"><a href="#11-translate-翻译DNA-RNA为蛋白质序列" class="headerlink" title="11. translate 翻译DNA/RNA为蛋白质序列"></a>11. translate 翻译DNA/RNA为蛋白质序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">--转化为蛋白序列</span></span><br><span class="line">seqkit translate gene.fa|head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 去除<span class="string">&#x27;X&#x27;</span> 和 <span class="string">&#x27;*&#x27;</span></span></span><br><span class="line">seqkit translate hairpin.fa </span><br><span class="line">seqkit translate hairpin.fa --trim | head</span><br></pre></td></tr></table></figure>

<h3 id="12-grep-序列匹配"><a href="#12-grep-序列匹配" class="headerlink" title="12. grep  序列匹配"></a>12. grep  序列匹配</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit grep [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">-n, --by-name # 匹配整个序列的名字，包含deion部分，而不是序列id。</span><br><span class="line">-s, --by-seq # 匹配序列</span><br><span class="line">-d, --degenerate</span><br><span class="line">pattern/motif 包含简并碱基</span><br><span class="line">-i, --ignore-case # 忽略大小写</span><br><span class="line">-v, --invert-match # 输出不匹配此模式的内容</span><br><span class="line">-p, # 匹配模式，支持连续写多个模式，匹配任一模式即输出。如-p ^ATG -p TAA$。注意该功能仅能正向匹配，不能实现对互补链匹配。</span><br><span class="line">-f, --pattern-file string # 支持匹配模式写到一个文件中，如要提取的序列ID。</span><br><span class="line">-R, --region string # 匹配位置选择。e.g 1:12 for first 12 bases, -12:-1 for last 12 bases</span><br><span class="line">-r, --use-regexp # 使用正则表达式，必须加入此参数,如^匹配首端。同-p联合使用。</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 生成一个ID列表</span></span><br><span class="line">grep &#x27;&gt;&#x27; C1_1.fa|cut -c2-|head -n 10 &gt; id.txt</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用序列id列表进行搜索(不包含空格)</span></span><br><span class="line">seqkit grep -f id.txt C1_1.fq.gz -o result.fq.gz</span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用序列名称列表进行搜索(它们可能包含空格)</span></span><br><span class="line">seqkit grep -n -f id.txt C1_1.fq.gz -o result.fa.gz</span><br><span class="line">zcat  result.fa.gz</span><br><span class="line"><span class="meta">#</span><span class="bash"> 提取hsa开头的序列</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -r -p ^hsa |head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> -v参数v用于移除序列</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -r -p ^hsa -p ^mmu -v | head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 提取ID</span></span><br><span class="line">zcat miRNA.diff.gz | grep ^# -v | grep NEW | cut -f 2 &gt; list</span><br><span class="line">head list</span><br><span class="line"><span class="meta">#</span><span class="bash"> 根据ID提取文件</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -f list &gt; new.fa</span><br><span class="line">head new.fa</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 提取包含特定碱基组合的序列</span></span><br><span class="line">cat hairpin.fa.gz | seqkit grep -s -i -p aggcg |head</span><br><span class="line"><span class="meta">#</span><span class="bash"> 统计</span></span><br><span class="line">cat hairpin.fa.gz | seqkit grep -s -i -p aggcg | seqkit stats</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash">  去除 包含特定组合碱基的序列</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit grep -s -r -i -p ^aggcg |head</span><br></pre></td></tr></table></figure>

<h3 id="13-locate-输出匹配位置"><a href="#13-locate-输出匹配位置" class="headerlink" title="13. locate 输出匹配位置"></a>13. locate 输出匹配位置</h3><p>对grep的拓展，可以正反链同时匹配，输出匹配的位置</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit locate [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">-d, --degenerate# pattern/motif contains degenerate base 包含简并碱基模式和motif</span><br><span class="line">--gtf # 输出为GTF格式</span><br><span class="line">-i, --ignore-case# ignore case 忽视大小写</span><br><span class="line">-P, --only-positive-strand# only search at positive strand 只在正链搜索</span><br><span class="line">-p, --pattern value# search pattern/motif</span><br><span class="line">-f, --pattern-file string# pattern/motif file (FASTA format) 模式或motif文件（fasta格式）</span><br><span class="line">-m, --max-mismatch int # 通过序列匹配时允许的最大错配</span><br><span class="line">-G, --non-greedy # 非贪婪模式，更快但是可能错过与其他重叠的motif</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">＃　其他两种输出格式</span><br><span class="line">zcat hairpin.fa.gz | seqkit locate -i -d -p AUGGACUN --bed</span><br><span class="line">zcat hairpin.fa.gz | seqkit locate -i -d -p AUGGACUN --gtf</span><br></pre></td></tr></table></figure>

<h3 id="14-fish-使用局部比对在较大的序列中寻找短序列"><a href="#14-fish-使用局部比对在较大的序列中寻找短序列" class="headerlink" title="14. fish 使用局部比对在较大的序列中寻找短序列"></a>14. fish 使用局部比对在较大的序列中寻找短序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">echo -e &#x27;&gt;seq\nACGACGACGA&#x27; \</span><br><span class="line">    | seqkit locate -p ACGA -G | csvtk -t pretty</span><br><span class="line"></span><br><span class="line">echo -e &#x27;&gt;seq\nACGACGACGA&#x27; \</span><br><span class="line">    | seqkit fish -F ACGA -a 2&gt;&amp;1 | csvtk -t pretty</span><br></pre></td></tr></table></figure>

<h3 id="15-amplicon-通过引物检索扩增子-或其周围的特定区域"><a href="#15-amplicon-通过引物检索扩增子-或其周围的特定区域" class="headerlink" title="15. amplicon 通过引物检索扩增子(或其周围的特定区域)"></a>15. amplicon 通过引物检索扩增子(或其周围的特定区域)</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 设置输出格式为bed，匹配的位点等信息</span></span><br><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt --bed</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash">- 使用引物文件，这用于刘老师组的高通量分菌的序列拆分速度应该很客观</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> cat seqs4amplicon.fa | seqkit amplicon -p primers.tsv --bed</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 输出除去引物之外的部分，-r输出几个碱基</span></span><br><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt -r 4:7</span><br><span class="line"><span class="meta">#</span><span class="bash"> 输出格式为bed</span></span><br><span class="line">echo -ne &quot;&gt;seq\nacgcccactgaaatga\n&quot; \</span><br><span class="line">    | seqkit amplicon -F ccc -R ttt -r 4:7 --bed</span><br></pre></td></tr></table></figure>

<h3 id="16-duplicate-对序列重复N次"><a href="#16-duplicate-对序列重复N次" class="headerlink" title="16. duplicate 对序列重复N次"></a>16. duplicate 对序列重复N次</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 重复序列1次，但是名字没有修改</span></span><br><span class="line">cat hairpin.fa | seqkit head -n 1 \</span><br><span class="line">    | seqkit duplicate -n 2</span><br><span class="line"><span class="meta">#</span><span class="bash"> 对重复序列改名，使其独一无二</span></span><br><span class="line">cat hairpin.fa | seqkit head -n 1 \</span><br><span class="line">    | seqkit duplicate -n 2 | seqkit rename</span><br></pre></td></tr></table></figure>

<h3 id="17-rmdup-通过id-名称-序列删除重复的序列"><a href="#17-rmdup-通过id-名称-序列删除重复的序列" class="headerlink" title="17. rmdup 通过id/名称/序列删除重复的序列"></a>17. rmdup 通过id/名称/序列删除重复的序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit rmdup [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">-n, --by-name 通过全名而不是id</span><br><span class="line">-s, --by-seq 通过序列</span><br><span class="line">-D, --dup-num-file string 保存数量的文件并列出重复的seqs</span><br><span class="line">-d, --dup-seqs-file string 保存重复seqs的文件</span><br><span class="line">-i, --ignore-case 忽视字母大小写</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 去除重复的序列</span></span><br><span class="line">zcat hairpin.fa.gz | seqkit rmdup -s -o clean.fa.gz</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 保存重复序列得到文件 -D duplicated.detail.txt</span></span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit rmdup -s -i -o clean.fa.gz -d duplicated.fa.gz -D duplicated.detail.txt</span><br></pre></td></tr></table></figure>

<h3 id="18-common-：通过id-名称-序列查找多个文件的公共序列"><a href="#18-common-：通过id-名称-序列查找多个文件的公共序列" class="headerlink" title="18. common ：通过id/名称/序列查找多个文件的公共序列"></a>18. common ：通过id/名称/序列查找多个文件的公共序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit common [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">-n, --by-name# 匹配整个序列的名字，包含deion部分，而不是序列id</span><br><span class="line">-s, --by-seq# match by sequence</span><br><span class="line">-i, --ignore-case# ignore case</span><br><span class="line">-m, --md5# use MD5 reduce memory usage</span><br></pre></td></tr></table></figure>

<p>这里同时支持fa和fq文件</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 通过ID匹配，文件夹下全部的fa序列公共部分输出来</span></span><br><span class="line">seqkit common *.fq.gz</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 通过-n实现全部名字匹配，-o输出结果</span></span><br><span class="line">seqkit common *.fq.gz -n -o common.fq </span><br><span class="line"><span class="meta">#</span><span class="bash"> 通过-s序列匹配</span></span><br><span class="line">seqkit common *.fq.gz -s -i -o common.fq</span><br></pre></td></tr></table></figure>

<h3 id="19-split-拆分序列为子文件"><a href="#19-split-拆分序列为子文件" class="headerlink" title="19. split 拆分序列为子文件"></a>19. split 拆分序列为子文件</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit split [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">-i, --by-id# split squences according to sequence ID</span><br><span class="line">-p, --by-part int# 将一个文件分割成N 份</span><br><span class="line">-s, --by-size int# 将一个文件按照N 条序列一个文件进行分割</span><br><span class="line">-O, --out-dir string# output directory (default value is infile.split)</span><br><span class="line">-2, --two-pass# two-pass mode to lower memory usage(only FAST)</span><br></pre></td></tr></table></figure>

<p>按名称ID、给定区域的子序列、文件大小或序列数量将序列拆分为文件</p>
<p>可用于将大文件拆分后，并行处理，加速分析。如从contig中预测基因。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">按照10000个序列为一个文件拆分，结果为hairpin.fa.gz.split/目录 ，文件名为hairpin.part_00x.fasta，-s</span></span><br><span class="line">seqkit split hairpin.fa.gz -s 10000 -2</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 将序列拆分为四个部分(常用，等分然后并行)</span></span><br><span class="line">seqkit split hairpin.fa.gz -p 5 -2</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 复杂一点的就是按照ID区分</span></span><br><span class="line">seqkit split hairpin.fa.gz -i --id-regexp &quot;^([\w]+)\-&quot; -2</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 按照前三个序列碱基来区分</span></span><br><span class="line">seqkit split hairpin.fa.gz -r 1:3 -2</span><br></pre></td></tr></table></figure>

<h3 id="20-split2-拆分文件-升级版本"><a href="#20-split2-拆分文件-升级版本" class="headerlink" title="20. split2 拆分文件 升级版本"></a>20. split2 拆分文件 升级版本</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">Flags:</span><br><span class="line">  -l, --by-length string   split sequences into chunks of &gt;=N bases, supports K/M/G suffix</span><br><span class="line">  -p, --by-part int        按照拆分出来的数量，比如：拆分成两个子文件2。-s, --by-size int        按照序列数量拆分</span><br><span class="line">  -f, --force              强制覆盖文件</span><br><span class="line">  -h, --help               查看帮助文件</span><br><span class="line">  -O, --out-dir string     输出文件夹 (default value is $infile.split)</span><br><span class="line">  -1, --read1 string       (gzipped) 双端序列第一个  </span><br><span class="line">  -2, --read2 string       (gzipped) 双端序列第二个</span><br></pre></td></tr></table></figure>

<p>同时支持fa和fq文件。单端和双端序列拆分实例</p>
<p>-f强制覆盖结果，适合重复计算时使用</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">seqkit split2 hairpin.fa.gz -s 10000 -f</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 双端序列拆分(重点)，p指定拆分数量，-O指定输出目录，-f覆盖结果，默认为压缩</span></span><br><span class="line">seqkit split2 -1 C1_1.fq.gz -2 C1_2.fq.gz -p 2 -O out -f</span><br></pre></td></tr></table></figure>

<h3 id="21-pair-拼接两个fastq文件"><a href="#21-pair-拼接两个fastq文件" class="headerlink" title="21. pair 拼接两个fastq文件"></a>21. pair 拼接两个fastq文件</h3><p>留下匹配的，去除不匹配的，这里我们使用扩增子的双端序列做一个演示：</p>
<p>注意：双端序列在两个文件中的顺序最好是一样的，否则会消耗大量内存去匹配。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">seqkit pair -1 C1_1.fq.gz -2 C1_2.fq.gz -O result</span><br><span class="line"><span class="meta">#</span><span class="bash"> -u 输出未匹配上的文件</span></span><br><span class="line">seqkit pair -1 C1_1.fq.gz -2 C1_2.fq.gz -O result -u -f</span><br></pre></td></tr></table></figure>

<h3 id="22-sample-按数量或比例对序列进行抽样。"><a href="#22-sample-按数量或比例对序列进行抽样。" class="headerlink" title="22. sample 按数量或比例对序列进行抽样。"></a>22. sample 按数量或比例对序列进行抽样。</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit sample [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">-n, --number int# sample by number (result may not exactly match)</span><br><span class="line">-p, --proportion float# sample by proportion（按比例提）</span><br><span class="line">-s, --rand-seed int# rand seed for shuffle (default 11)</span><br><span class="line">-2, --two-pass# 2-pass modelower memory</span><br></pre></td></tr></table></figure>

<p>按照百分比例和序列数量进行抽样</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 抽样百分之十</span></span><br><span class="line">zcat C1_1.fq.gz | seqkit sample -p 0.1 -o sample.fq.gz</span><br><span class="line"><span class="meta">#</span><span class="bash"> 抽样1000条</span></span><br><span class="line">zcat C1_1.fq.gz | seqkit sample -n 1000 -o sample.fq.gz</span><br></pre></td></tr></table></figure>

<p>注意：1000条并不是很准确，可能是900多条，为什么呢？看这里了解问题。<a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/note/#effect-of-random-seed-on-results-of-seqkit-sample">https://bioinf.shenwei.me/seqkit/note/#effect-of-random-seed-on-results-of-seqkit-sample</a></p>
<p>这里为大家展示一下减少内存的序列抽样方法</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 抽样 seqkit sample</span> </span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit sample -p 0.1 \</span><br><span class="line">    | seqkit head -n 1000 -o sample.fa.gz</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 设置随机种子，方便重复结果: -s 11</span></span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit sample -p 0.1 -s 11 |head</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 抽样后打乱序列 :seqkit shuffle</span></span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit sample -p 0.1 \</span><br><span class="line">    | seqkit shuffle -o sample.fa.gz</span><br></pre></td></tr></table></figure>

<h3 id="23-range-打印序列-按照一个范围"><a href="#23-range-打印序列-按照一个范围" class="headerlink" title="23. range 打印序列 按照一个范围"></a>23. range 打印序列 按照一个范围</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 打印一个范围内的序列</span></span><br><span class="line">cat hairpin.fa | seqkit range -r 1:10</span><br><span class="line"><span class="meta">#</span><span class="bash"> 打印最后几行的序列</span></span><br><span class="line">cat hairpin.fa | seqkit range -r -100:-1 | seqkit stats</span><br><span class="line"><span class="meta">#</span><span class="bash"> 打印中间范围的序列</span></span><br><span class="line">cat hairpin.fa | seqkit range -r 101:150 | seqkit stats</span><br></pre></td></tr></table></figure>

<h3 id="24-replace-使用正则表达式替换名称-序列。"><a href="#24-replace-使用正则表达式替换名称-序列。" class="headerlink" title="24. replace 使用正则表达式替换名称/序列。"></a>24. replace 使用正则表达式替换名称/序列。</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit replace [flag]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">-s, --by-seq # 代替seq</span><br><span class="line">-i, --ignore-case # 忽视大小写</span><br><span class="line">-K, --keep-key </span><br><span class="line">-p, --pattern # string 搜索正则表达式</span><br><span class="line">-r, --replacement # string 替换物从两个配对端读数的文件提取配对的reads</span><br></pre></td></tr></table></figure>

<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 修改序列名称：删除空格后内存</span></span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot;\s.+&quot;</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 修改序列名：替换</span></span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot;\-&quot; -r &#x27;=&#x27;</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 修改序列：去除序列间隔</span></span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot; |-&quot; -s</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 修改序列：给每一个碱基加上空格</span></span><br><span class="line">echo -e &quot;&gt;seq1 abc-123\nACGT-ACGT&quot; \</span><br><span class="line">    | seqkit replace -p &quot;(.)&quot; -r &#x27;$1 &#x27; -s</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 使用字符加数据重命名序列-用于扩增子代表序列改名非常优秀</span></span><br><span class="line">echo -e &quot;&gt;abc\nACTG\n&gt;123\nATTT&quot; \</span><br><span class="line">    |  seqkit replace -p .+ -r &quot;ASV_&#123;nr&#125;&quot;</span><br><span class="line"></span><br><span class="line">echo -e &quot;&gt;abc\nACTG\n&gt;123\nATTT&quot; \</span><br><span class="line">    |  seqkit replace -p .+ -r &quot;SAV_&#123;nr&#125;&quot; --nr-width 5</span><br></pre></td></tr></table></figure>

<h3 id="25-rename-重命名重复的ID"><a href="#25-rename-重命名重复的ID" class="headerlink" title="25. rename 重命名重复的ID"></a>25. rename 重命名重复的ID</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> 重命名：相同序列会在后面加上_2 来处理</span></span><br><span class="line">echo -e &quot;&gt;a comment\nacgt\n&gt;b comment of b\nACTG\n&gt;a comment\naaaa&quot; \</span><br><span class="line">    | seqkit rename</span><br></pre></td></tr></table></figure>

<h3 id="26-concat-连接多个文件中具有相同ID的序列"><a href="#26-concat-连接多个文件中具有相同ID的序列" class="headerlink" title="26. concat 连接多个文件中具有相同ID的序列"></a>26. concat 连接多个文件中具有相同ID的序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash">这里演示组合前面两个碱基和最后两个碱基的用法</span></span><br><span class="line">seqkit concat &lt;(seqkit subseq -r 1:2 C1_1.fq.gz) &lt;(seqkit subseq -r -2:-1 C1_2.fq.gz)|head</span><br></pre></td></tr></table></figure>

<h3 id="27-shuffle-随机打乱序列-默认全部读入内存"><a href="#27-shuffle-随机打乱序列-默认全部读入内存" class="headerlink" title="27. shuffle 随机打乱序列 默认全部读入内存"></a>27. shuffle 随机打乱序列 默认全部读入内存</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit shuffle hairpin.fa.gz -2  &gt; shuffled.fa</span><br></pre></td></tr></table></figure>

<h3 id="28-sort-按id-名称-序列-长度排序序列"><a href="#28-sort-按id-名称-序列-长度排序序列" class="headerlink" title="28. sort 按id/名称/序列/长度排序序列"></a>28. sort 按id/名称/序列/长度排序序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit sort [flags]</span><br></pre></td></tr></table></figure>

<p><strong>参数</strong></p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">-l, --by-length# 按照序列长度排序</span><br><span class="line">-n, --by-name# by full name</span><br><span class="line">-s, --by-seq# 按照序列排序</span><br><span class="line">-i, --ignore-case# 按序列排序时忽略大小写</span><br><span class="line">-r, --reverse# 反向排序</span><br><span class="line">-2, --two-pass# 对于FASTA序列排序可以减少内存</span><br></pre></td></tr></table></figure>

<ul>
<li>—quiet 屏幕不输出过程</li>
<li>-i 排序忽略大小写</li>
<li>-l 按照序列长度排序</li>
</ul>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> ID排序</span></span><br><span class="line">echo -e &quot;&gt;seq1\nACGTNcccc\n&gt;SEQ2\nacgtnAAAA&quot; \</span><br><span class="line">    | seqkit sort --quiet</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 按照ID排序，忽略大小写</span></span><br><span class="line">echo -e &quot;&gt;seq1\nACGTNcccc\n&gt;SEQ2\nacgtnAAAA&quot; \</span><br><span class="line">    | seqkit sort --quiet -i</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 按照序列长度排序，由小到大</span></span><br><span class="line">echo -e &quot;&gt;seq1\nACGTNcccc\n&gt;SEQ2\nacgtnAAAAnnn\n&gt;seq3\nacgt&quot; \</span><br><span class="line">    | seqkit sort --quiet -l</span><br></pre></td></tr></table></figure>

<h2 id="其他功能"><a href="#其他功能" class="headerlink" title="其他功能"></a>其他功能</h2><h3 id="1-mutate-编辑序列-点突变、插入、删除"><a href="#1-mutate-编辑序列-点突变、插入、删除" class="headerlink" title="1. mutate 编辑序列(点突变、插入、删除)"></a>1. mutate 编辑序列(点突变、插入、删除)</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot;</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 修改第一个碱基</span></span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -p 1:x</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 修改第五个位置的碱基，输出信息隐藏</span>    </span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -p 5:x --quiet</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 可以同时修改多个碱基</span></span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -p 1:x -p -1:x --quiet</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 删除碱基</span></span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -d 1:1 --quiet</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 删除倒数三个碱基</span></span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -d -3:-1 --quiet</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 插入碱基</span></span><br><span class="line">echo -ne &quot;&gt;1\nACTGNactgn\n&gt;2\nactgnACTGN\n&quot; \</span><br><span class="line">    | seqkit mutate -i 0:xx --quiet</span><br></pre></td></tr></table></figure>

<h3 id="2-head-展示开头N行的序列"><a href="#2-head-展示开头N行的序列" class="headerlink" title="2. head 展示开头N行的序列"></a>2. head 展示开头N行的序列</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit head -n 1 hairpin.fa.gz</span><br></pre></td></tr></table></figure>

<h3 id="3-locate-定位子序列或者保守序列位置"><a href="#3-locate-定位子序列或者保守序列位置" class="headerlink" title="3. locate 定位子序列或者保守序列位置"></a>3. locate 定位子序列或者保守序列位置</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">cat gene.fa | seqkit locate -p ACT | csvtk pretty -t</span><br><span class="line"><span class="meta">#</span><span class="bash"> 调整错配 最大错配为1</span></span><br><span class="line">cat gene.fa \</span><br><span class="line">  | seqkit locate -p ACTG -m 1 \</span><br><span class="line">  | csvtk pretty -t</span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 简并碱基</span></span><br><span class="line">zcat hairpin.fa.gz \</span><br><span class="line">    | seqkit locate -i -d -p AUGGACUN \</span><br><span class="line">    | head -n 4</span><br></pre></td></tr></table></figure>

<h3 id="4-restart-重置循环基因组的起始位置"><a href="#4-restart-重置循环基因组的起始位置" class="headerlink" title="4. restart 重置循环基因组的起始位置"></a>4. restart 重置循环基因组的起始位置</h3><p>这个使用的比较少</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">echo -e &quot;&gt;seq\nacgtnACGTN&quot;</span><br><span class="line"></span><br><span class="line">echo -e &quot;&gt;seq\nacgtnACGTN&quot; | seqkit restart -i 2</span><br><span class="line">echo -e &quot;&gt;seq\nacgtnACGTN&quot; | seqkit restart -i -2</span><br></pre></td></tr></table></figure>

<h3 id="5-convet-二代测序质量值的转化为-Sanger"><a href="#5-convet-二代测序质量值的转化为-Sanger" class="headerlink" title="5. convet 二代测序质量值的转化为 Sanger"></a>5. convet 二代测序质量值的转化为 Sanger</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#</span><span class="bash"> Illumina-1.8+ -&gt; Sanger</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> seqkit convert tests/Illimina1.8.fq.gz  | seqkit head -n 1</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> 40分以上的都认为40 Illumina-1.8+ -&gt; Sanger</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> seqkit convert tests/Illimina1.8.fq.gz -f | seqkit head -n 1</span></span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash">Illumina-1.8+ -&gt; Illumina-1.5+</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> seqkit convert tests/Illimina1.8.fq.gz --to Illumina-1.5+ | seqkit head -n 1</span></span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> Illumina-1.8+ -&gt; Illumina-1.5+ -&gt;  Sanger.</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> seqkit convert tests/Illimina1.8.fq.gz --to Illumina-1.5+ | seqkit convert | seqkit head -n 1</span></span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> Solexa -&gt; Sanger</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> seqkit convert tests/Illimina1.8.fq.gz --from Solexa</span></span><br><span class="line"><span class="meta"></span></span><br><span class="line"><span class="meta">#</span><span class="bash"> Illumina-1.5+ -&gt; Sanger</span></span><br><span class="line"><span class="meta">#</span><span class="bash"> seqkit convert tests/Illimina1.5.fq | seqkit head -n 1</span></span><br></pre></td></tr></table></figure>

<h2 id="技术细节和使用"><a href="#技术细节和使用" class="headerlink" title="技术细节和使用"></a>技术细节和使用</h2><h3 id="1-seqkit处理压缩文件"><a href="#1-seqkit处理压缩文件" class="headerlink" title="1. seqkit处理压缩文件"></a>1. seqkit处理压缩文件</h3><p>pigz 或者 gzip 在部分操作中不能加速，所以在v.0.8.1版本以便关注了，然而还是可以使用命令：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">pigz -d -c seqs.fq.gz | seqkit xxx</span><br></pre></td></tr></table></figure>

<p>因为seqkit使用了pgzip去写gzip。这比gzip和pigz更快。（10X of gzip, 4X of pigz），而且gzip压缩文件比较大。</p>
<h3 id="2-从数据处理格式来讲"><a href="#2-从数据处理格式来讲" class="headerlink" title="2. 从数据处理格式来讲"></a>2. 从数据处理格式来讲</h3><p>​    seqkit无缝支持fa和fq格式数据，并且可以自动识别。除了 faidx之外，全部命令都可以处理这两种格式的数据。</p>
<p>​    只有fa格式支持命令(subseq, split, sort和shuffle)利用FASTA索引(by flag -two-pass)下提高大文件的性能。</p>
<p>​    序列类型的检测DNA/RNA/Protein，会使用子序列进行，默认检测第一条子序列，通过—alphabet-guess-seq-length参数默认为10000，如果长度小于10000，则检查整条序列。</p>
<h3 id="3-序列名字"><a href="#3-序列名字" class="headerlink" title="3. 序列名字"></a>3. 序列名字</h3><p>​    所有的软件，包括seqkit，<strong>使用第一个空格之前的字符作为序列的名字</strong>：</p>
<p>​    需要注意的NCBI等一些序列的格式并不是如此，例如：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">&gt;</span><span class="bash">gi|110645304|ref|NC_002516.2| Pseudomona</span></span><br></pre></td></tr></table></figure>

<p>​    想要在seqkit中识别出来的序列ID为：NC_002516.2。</p>
<p>​    此时使用参数<code>--id-regexp &quot;\|([^\|]+)\| &quot;</code>，或者添加参数<code>--id-ncbi</code>，但如果是只要前面的gi数字作为ID的话，添加参数：<code>--id-regexp &quot;^gi\|([^\|]+)\|&quot;</code>。</p>
<p>​    注意：.seqkit.fai不同于samtools产生的.fai格式文件，seqkit使用整个序列开头而不是ID作为索引。</p>
<h3 id="4-并行运算"><a href="#4-并行运算" class="headerlink" title="4. 并行运算"></a>4. 并行运算</h3><p>​    单核CPU默认线程：—threads 1，多个CPU，线程默认2.</p>
<h3 id="5-内存占用"><a href="#5-内存占用" class="headerlink" title="5. 内存占用"></a>5. 内存占用</h3><p>​    seqkit许多的命令都不需要将整个序列读入到内存中。包括：stat, fq2fa, fx2tab, tab2fx, grep, locate, replace, seq, sliding, subseq。</p>
<p>​    注意：如果使用subseq —gtf | —bed时，如果GTF或者BED文件太大，内存使用量会暴增，可以通过指定染色体：—chr，或者—feature去限制特征。</p>
<p>​    有一些命令需要将文件读入内存，但是可以用过rmdup 和 common减少内存使用。</p>
<h3 id="6-随机—抽样"><a href="#6-随机—抽样" class="headerlink" title="6. 随机—抽样"></a>6. 随机—抽样</h3><p>​    抽样命令sample和shuffle使用了随机功能，为了保证重现性，可以使用<code>-s</code>设置随机种子。这可以保证在不同的环境中可以有相同的抽样结果。</p>

    </div>
    
    <div>
      
          <div>
    
        <div style="text-align:center;color: #ccc;font-size:14px;">
              -------------本文结束<i class="fa fa-paw"></i>感谢您的阅读-------------
        </div>
    
</div>


      
    </div>

    
    
    
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>本文作者： </strong>阿狒
  </li>
  <li class="post-copyright-link">
    <strong>本文链接：</strong>
    <a href="http://zhengxingsong.github.io/%E7%94%9F%E4%BF%A1/seqkit/" title="seqkit">http://zhengxingsong.github.io/生信/seqkit/</a>
  </li>
  <li class="post-copyright-license">
    <strong>版权声明： </strong>本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/zh-CN" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> 许可协议。转载请注明出处！
  </li>
</ul>
</div>

        


      <footer class="post-footer">
          
          <div class="post-tags">
              <a href="/tags/%E7%94%9F%E4%BF%A1/" rel="tag"><i class="fa fa-tag"></i> 生信</a>
              <a href="/tags/biosoft/" rel="tag"><i class="fa fa-tag"></i> biosoft</a>
              <a href="/tags/CommonSoft/" rel="tag"><i class="fa fa-tag"></i> CommonSoft</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/%E7%94%9F%E4%BF%A1/csvtk/" rel="prev" title="csvtk">
      <i class="fa fa-chevron-left"></i> csvtk
    </a></div>
      <div class="post-nav-item">
    <a href="/Docker/Docker-%E5%B8%B8%E7%94%A8%E5%91%BD%E4%BB%A4/" rel="next" title="Docker 常用命令">
      Docker 常用命令 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%AE%89%E8%A3%85"><span class="nav-number">1.</span> <span class="nav-text">安装</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%B8%B8%E7%94%A8%E5%8A%9F%E8%83%BD"><span class="nav-number">2.</span> <span class="nav-text">常用功能</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-%E6%A0%B9%E6%8D%AEgtf%E6%9E%84%E5%BB%BAbed%E6%96%87%E4%BB%B6"><span class="nav-number">2.1.</span> <span class="nav-text">1.根据gtf构建bed文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-stats-FASTA-Q%E6%96%87%E4%BB%B6%E7%9A%84%E7%AE%80%E5%8D%95%E7%BB%9F%E8%AE%A1"><span class="nav-number">2.2.</span> <span class="nav-text">2. stats FASTA&#x2F;Q文件的简单统计</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-seq-%E8%BD%AC%E6%8D%A2%E5%BA%8F%E5%88%97-%EF%BC%88%E5%8F%8D%E5%90%91%E3%80%81%E4%BA%92%E8%A1%A5-%E6%8F%90%E5%8F%96ID%EF%BC%89"><span class="nav-number">2.3.</span> <span class="nav-text">3. seq 转换序列 （反向、互补&#x2F;提取ID）</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#3-1-%E6%8C%89%E9%95%BF%E5%BA%A6%E8%BF%87%E6%BB%A4-%E5%B8%B8%E7%94%A8"><span class="nav-number">2.3.1.</span> <span class="nav-text">3.1 按长度过滤(常用)</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#3-2-%E6%8F%90%E5%8F%96ID"><span class="nav-number">2.3.2.</span> <span class="nav-text">3.2 提取ID</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#3-3-%E5%8D%95%E8%A1%8C-%E5%A4%9A%E8%A1%8C%E8%BD%AC%E6%8D%A2"><span class="nav-number">2.3.3.</span> <span class="nav-text">3.3 单行&#x2F;多行转换</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#3-4-%E5%8F%8D%E5%90%91-%E4%BA%92%E8%A1%A5"><span class="nav-number">2.3.4.</span> <span class="nav-text">3.4 反向&#x2F;互补</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#3-5-%E5%88%A0%E9%99%A4gap-%E5%A4%A7%E5%B0%8F%E5%86%99%E8%BD%AC%E6%8D%A2"><span class="nav-number">2.3.5.</span> <span class="nav-text">3.5 删除gap&#x2F;大小写转换</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#3-6-RNA%E8%BD%AC%E4%B8%BADNA"><span class="nav-number">2.3.6.</span> <span class="nav-text">3.6 RNA转为DNA</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-subseq%E9%80%9A%E8%BF%87%E6%8C%87%E5%AE%9A%E5%8C%BA%E5%9F%9F"><span class="nav-number">2.4.</span> <span class="nav-text">4. subseq通过指定区域</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-sliding-%E6%BB%91%E7%AA%97%E6%8F%90%E5%8F%96%E5%BA%8F%E5%88%97%EF%BC%8C%E6%94%AF%E6%8C%81%E7%8E%AF%E7%8A%B6%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">2.5.</span> <span class="nav-text">5. sliding 滑窗提取序列，支持环状基因组</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-faidx-%E5%88%9B%E5%BB%BAFASTA%E7%B4%A2%E5%BC%95%E6%96%87%E4%BB%B6%E5%B9%B6%E6%8F%90%E5%8F%96%E5%AD%90%E5%BA%8F%E5%88%97"><span class="nav-number">2.6.</span> <span class="nav-text">6. faidx 创建FASTA索引文件并提取子序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#7-watch-%E5%BA%8F%E5%88%97%E8%B4%A8%E9%87%8F%E7%9A%84%E7%9B%91%E6%B5%8B%E5%92%8C%E5%9C%A8%E7%BA%BF%E7%9B%B4%E6%96%B9%E5%9B%BE"><span class="nav-number">2.7.</span> <span class="nav-text">7. watch 序列质量的监测和在线直方图</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#8-sana%EF%BC%9A%E6%B8%85%E7%90%86%E6%8D%9F%E5%9D%8Ffastq%E6%96%87%E4%BB%B6"><span class="nav-number">2.8.</span> <span class="nav-text">8. sana：清理损坏fastq文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#9-fq2fa-%E5%B0%86fq%E8%BD%AC%E4%B8%BAfa%E6%A0%BC%E5%BC%8F"><span class="nav-number">2.9.</span> <span class="nav-text">9. fq2fa 将fq转为fa格式</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#10-fx2tab-amp-tab2fx-%E5%BA%8F%E5%88%97%E8%BD%AC%E5%8C%96%E8%A1%A8%E6%A0%BC%E6%A0%BC%E5%BC%8F"><span class="nav-number">2.10.</span> <span class="nav-text">10. fx2tab &amp; tab2fx 序列转化表格格式</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#11-translate-%E7%BF%BB%E8%AF%91DNA-RNA%E4%B8%BA%E8%9B%8B%E7%99%BD%E8%B4%A8%E5%BA%8F%E5%88%97"><span class="nav-number">2.11.</span> <span class="nav-text">11. translate 翻译DNA&#x2F;RNA为蛋白质序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#12-grep-%E5%BA%8F%E5%88%97%E5%8C%B9%E9%85%8D"><span class="nav-number">2.12.</span> <span class="nav-text">12. grep  序列匹配</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#13-locate-%E8%BE%93%E5%87%BA%E5%8C%B9%E9%85%8D%E4%BD%8D%E7%BD%AE"><span class="nav-number">2.13.</span> <span class="nav-text">13. locate 输出匹配位置</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#14-fish-%E4%BD%BF%E7%94%A8%E5%B1%80%E9%83%A8%E6%AF%94%E5%AF%B9%E5%9C%A8%E8%BE%83%E5%A4%A7%E7%9A%84%E5%BA%8F%E5%88%97%E4%B8%AD%E5%AF%BB%E6%89%BE%E7%9F%AD%E5%BA%8F%E5%88%97"><span class="nav-number">2.14.</span> <span class="nav-text">14. fish 使用局部比对在较大的序列中寻找短序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#15-amplicon-%E9%80%9A%E8%BF%87%E5%BC%95%E7%89%A9%E6%A3%80%E7%B4%A2%E6%89%A9%E5%A2%9E%E5%AD%90-%E6%88%96%E5%85%B6%E5%91%A8%E5%9B%B4%E7%9A%84%E7%89%B9%E5%AE%9A%E5%8C%BA%E5%9F%9F"><span class="nav-number">2.15.</span> <span class="nav-text">15. amplicon 通过引物检索扩增子(或其周围的特定区域)</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#16-duplicate-%E5%AF%B9%E5%BA%8F%E5%88%97%E9%87%8D%E5%A4%8DN%E6%AC%A1"><span class="nav-number">2.16.</span> <span class="nav-text">16. duplicate 对序列重复N次</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#17-rmdup-%E9%80%9A%E8%BF%87id-%E5%90%8D%E7%A7%B0-%E5%BA%8F%E5%88%97%E5%88%A0%E9%99%A4%E9%87%8D%E5%A4%8D%E7%9A%84%E5%BA%8F%E5%88%97"><span class="nav-number">2.17.</span> <span class="nav-text">17. rmdup 通过id&#x2F;名称&#x2F;序列删除重复的序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#18-common-%EF%BC%9A%E9%80%9A%E8%BF%87id-%E5%90%8D%E7%A7%B0-%E5%BA%8F%E5%88%97%E6%9F%A5%E6%89%BE%E5%A4%9A%E4%B8%AA%E6%96%87%E4%BB%B6%E7%9A%84%E5%85%AC%E5%85%B1%E5%BA%8F%E5%88%97"><span class="nav-number">2.18.</span> <span class="nav-text">18. common ：通过id&#x2F;名称&#x2F;序列查找多个文件的公共序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#19-split-%E6%8B%86%E5%88%86%E5%BA%8F%E5%88%97%E4%B8%BA%E5%AD%90%E6%96%87%E4%BB%B6"><span class="nav-number">2.19.</span> <span class="nav-text">19. split 拆分序列为子文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#20-split2-%E6%8B%86%E5%88%86%E6%96%87%E4%BB%B6-%E5%8D%87%E7%BA%A7%E7%89%88%E6%9C%AC"><span class="nav-number">2.20.</span> <span class="nav-text">20. split2 拆分文件 升级版本</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#21-pair-%E6%8B%BC%E6%8E%A5%E4%B8%A4%E4%B8%AAfastq%E6%96%87%E4%BB%B6"><span class="nav-number">2.21.</span> <span class="nav-text">21. pair 拼接两个fastq文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#22-sample-%E6%8C%89%E6%95%B0%E9%87%8F%E6%88%96%E6%AF%94%E4%BE%8B%E5%AF%B9%E5%BA%8F%E5%88%97%E8%BF%9B%E8%A1%8C%E6%8A%BD%E6%A0%B7%E3%80%82"><span class="nav-number">2.22.</span> <span class="nav-text">22. sample 按数量或比例对序列进行抽样。</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#23-range-%E6%89%93%E5%8D%B0%E5%BA%8F%E5%88%97-%E6%8C%89%E7%85%A7%E4%B8%80%E4%B8%AA%E8%8C%83%E5%9B%B4"><span class="nav-number">2.23.</span> <span class="nav-text">23. range 打印序列 按照一个范围</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#24-replace-%E4%BD%BF%E7%94%A8%E6%AD%A3%E5%88%99%E8%A1%A8%E8%BE%BE%E5%BC%8F%E6%9B%BF%E6%8D%A2%E5%90%8D%E7%A7%B0-%E5%BA%8F%E5%88%97%E3%80%82"><span class="nav-number">2.24.</span> <span class="nav-text">24. replace 使用正则表达式替换名称&#x2F;序列。</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#25-rename-%E9%87%8D%E5%91%BD%E5%90%8D%E9%87%8D%E5%A4%8D%E7%9A%84ID"><span class="nav-number">2.25.</span> <span class="nav-text">25. rename 重命名重复的ID</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#26-concat-%E8%BF%9E%E6%8E%A5%E5%A4%9A%E4%B8%AA%E6%96%87%E4%BB%B6%E4%B8%AD%E5%85%B7%E6%9C%89%E7%9B%B8%E5%90%8CID%E7%9A%84%E5%BA%8F%E5%88%97"><span class="nav-number">2.26.</span> <span class="nav-text">26. concat 连接多个文件中具有相同ID的序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#27-shuffle-%E9%9A%8F%E6%9C%BA%E6%89%93%E4%B9%B1%E5%BA%8F%E5%88%97-%E9%BB%98%E8%AE%A4%E5%85%A8%E9%83%A8%E8%AF%BB%E5%85%A5%E5%86%85%E5%AD%98"><span class="nav-number">2.27.</span> <span class="nav-text">27. shuffle 随机打乱序列 默认全部读入内存</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#28-sort-%E6%8C%89id-%E5%90%8D%E7%A7%B0-%E5%BA%8F%E5%88%97-%E9%95%BF%E5%BA%A6%E6%8E%92%E5%BA%8F%E5%BA%8F%E5%88%97"><span class="nav-number">2.28.</span> <span class="nav-text">28. sort 按id&#x2F;名称&#x2F;序列&#x2F;长度排序序列</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%85%B6%E4%BB%96%E5%8A%9F%E8%83%BD"><span class="nav-number">3.</span> <span class="nav-text">其他功能</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-mutate-%E7%BC%96%E8%BE%91%E5%BA%8F%E5%88%97-%E7%82%B9%E7%AA%81%E5%8F%98%E3%80%81%E6%8F%92%E5%85%A5%E3%80%81%E5%88%A0%E9%99%A4"><span class="nav-number">3.1.</span> <span class="nav-text">1. mutate 编辑序列(点突变、插入、删除)</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-head-%E5%B1%95%E7%A4%BA%E5%BC%80%E5%A4%B4N%E8%A1%8C%E7%9A%84%E5%BA%8F%E5%88%97"><span class="nav-number">3.2.</span> <span class="nav-text">2. head 展示开头N行的序列</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-locate-%E5%AE%9A%E4%BD%8D%E5%AD%90%E5%BA%8F%E5%88%97%E6%88%96%E8%80%85%E4%BF%9D%E5%AE%88%E5%BA%8F%E5%88%97%E4%BD%8D%E7%BD%AE"><span class="nav-number">3.3.</span> <span class="nav-text">3. locate 定位子序列或者保守序列位置</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-restart-%E9%87%8D%E7%BD%AE%E5%BE%AA%E7%8E%AF%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%9A%84%E8%B5%B7%E5%A7%8B%E4%BD%8D%E7%BD%AE"><span class="nav-number">3.4.</span> <span class="nav-text">4. restart 重置循环基因组的起始位置</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-convet-%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8F%E8%B4%A8%E9%87%8F%E5%80%BC%E7%9A%84%E8%BD%AC%E5%8C%96%E4%B8%BA-Sanger"><span class="nav-number">3.5.</span> <span class="nav-text">5. convet 二代测序质量值的转化为 Sanger</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%8A%80%E6%9C%AF%E7%BB%86%E8%8A%82%E5%92%8C%E4%BD%BF%E7%94%A8"><span class="nav-number">4.</span> <span class="nav-text">技术细节和使用</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-seqkit%E5%A4%84%E7%90%86%E5%8E%8B%E7%BC%A9%E6%96%87%E4%BB%B6"><span class="nav-number">4.1.</span> <span class="nav-text">1. seqkit处理压缩文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-%E4%BB%8E%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86%E6%A0%BC%E5%BC%8F%E6%9D%A5%E8%AE%B2"><span class="nav-number">4.2.</span> <span class="nav-text">2. 从数据处理格式来讲</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-%E5%BA%8F%E5%88%97%E5%90%8D%E5%AD%97"><span class="nav-number">4.3.</span> <span class="nav-text">3. 序列名字</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-%E5%B9%B6%E8%A1%8C%E8%BF%90%E7%AE%97"><span class="nav-number">4.4.</span> <span class="nav-text">4. 并行运算</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-%E5%86%85%E5%AD%98%E5%8D%A0%E7%94%A8"><span class="nav-number">4.5.</span> <span class="nav-text">5. 内存占用</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-%E9%9A%8F%E6%9C%BA%E2%80%94%E6%8A%BD%E6%A0%B7"><span class="nav-number">4.6.</span> <span class="nav-text">6. 随机—抽样</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="阿狒"
      src="/images/header.jpg">
  <p class="site-author-name" itemprop="name">阿狒</p>
  <div class="site-description" itemprop="description">人这一生总会死去，你不妨大胆一些，爱一个人，攀一座山，追一个梦，是的，你不妨大胆一些。</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">7</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">4</span>
        <span class="site-state-item-name">分类</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">9</span>
        <span class="site-state-item-name">标签</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/zhengxingSong" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;zhengxingSong" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">阿狒</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="站点总字数">77k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="站点阅读时长">1:10</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="总访客量">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="总访问量">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>

<script src="/js/bookmark.js"></script>


  <script defer src="/lib/three/three.min.js"></script>
    <script defer src="/lib/three/three-waves.min.js"></script>


  















  

  

</body>
</html>
